e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js

```diff
@@ -1103,3 +1103,84 @@ describe("issue 49342", () => {
     cy.button("Cancel").should("be.focused");
   });
 });
+
+describe("issue 49882", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should not eat up subsequent characters when applying a suggestion (metabase#49882-1)", () => {
+    openOrdersTable({ mode: "notebook" });
+    cy.findByLabelText("Custom column").click();
+
+    const moveCursorTo2ndCaseArgument = "{leftarrow}".repeat(6);
+    enterCustomColumnDetails({
+      formula: `case([Total] > 200, , "X")${moveCursorTo2ndCaseArgument}[to{enter}`,
+    });
+
+    cy.get(".ace_text-input")
+      .first()
+      .should("have.value", 'case([Total] > 200, [Total] , "X")\n\n');
```

Comment:
You may wonder why there's a space after the second `[Total]`.
A space is always added for suggestions that are not functions (for functions `(` is added instead): https://github.com/metabase/metabase/blob/92d615073a3ba31710e26be9f2e83141352d1d1c/frontend/src/metabase-lib/v1/expressions/suggest.ts#L42-L44

I think we could skip the space when the token is followed by an operator, but just to not make this PR any larger, I left it out.

Anyway, this is just a cosmetic things since blurring the expression input auto-formats the expression.